### PR TITLE
fix: prevent possible overflow

### PR DESCRIPTION
**Pull request by armcknight**
https://github.com/getsentry/sentry-cocoa/security/code-scanning/1

#skip-changelog

### CODE REVIEW

**Review by brustolin — APPROVED** [review]

File: Sources/SentryCrash/Recording/Tools/SentryCrashObjC.c [hunk 1]
```diff
@@ -1122,7 +1122,8 @@ sentrycrashobjc_ivarValue(const void *const objectPtr, int ivarIndex, void *dst)
         return false;
     }
     uintptr_t ivarPtr = (uintptr_t)&ivars->first;
-    const struct ivar_t *ivar = (void *)(ivarPtr + ivars->entsizeAndFlags * (unsigned)ivarIndex);
+    const struct ivar_t *ivar
```
Comment (philipphofmann) [inline Sources/SentryCrash/Recording/Tools/SentryCrashObjC.c:1125]: `m`: Can we add a test for that in `SentryCrashObjC_Tests` please?

**Comment by armcknight** [comment]
Opened a PR upstream: https://github.com/kstenerud/KSCrash/pull/430

**Comment by philipphofmann** [comment]
> Will give maintainers there a while to comment before moving forward with this

It can be that it takes months until you get feedback @armcknight. I opened a simple PR on the first of April and didn't get any feedback yet, see https://github.com/kstenerud/KSCrash/pull/427.

**Comment by philipphofmann** [comment]
We can also ask @Swatinem to have a look at this tiny change. @Swatinem, do you think this makes sense?

**Review by Swatinem — APPROVED** [review]

**Pull request merged**